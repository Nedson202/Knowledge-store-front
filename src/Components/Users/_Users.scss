/*.admin-panel {
  display: grid;
  grid-template-columns: 1fr 3fr;
  max-width: 100vw;
  height: 100vh;
  margin: 55px auto -110px auto;
  border: 1px solid #005C97;
}

@media screen and (max-width: 450px) {
	.add-book-form {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    max-width: 800px;
    margin: 0 auto;
    grid-gap: 20px;
  }
}*/

.admin-panel {
  margin: 120px auto 0 auto;
  max-width: 950px;
  width: 100%;

  &__header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;
  }

  .ant-select, .ant-input {
    width: 120px !important;
    margin-right: 15px;
    height: 36px !important;
  }

  .ant-select, .ant-input:focus {
    outline: 0;
    box-shadow: none;
    border-color: transparent;
  }

  .table-spinner {
    position: absolute;
    max-width: 100px;
    width: 100%;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
  }
}

@media (max-width: 500px) {
  .admin-panel {
    width: 90% !important;
    margin: 120px 20px 0 0px !important;

    &__header {
      width: 105%;

      h3 {
        
      }
    }
  }
}