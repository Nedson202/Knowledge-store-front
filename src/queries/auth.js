import { gql } from 'apollo-boost';

const addUser = gql`
  mutation($username: String!, $password: String!, $email: String!) {
    addUser(username: $username, password: $password, email: $email) {
      token
    }
  }
`;

const verifyEmail = gql`
  mutation($token: String!) {
    verifyEmail(token: $token) {
      token,
      message
    }
  }
`;

const sendVerificationEmail = gql`
  query($email: String!) {
    sendVerificationEmail(email: $email) {
      message
    }
  }
`;

const forgotPassword = gql`
  query($email: String!) {
    forgotPassword(email: $email) {
      message
    }
  }
`;

const resetPassword = gql`
  mutation($id: String!, $email: String!, $password: String!, $token: String!) {
    resetPassword(id: $id, email: $email, password: $password, token: $token) {
      message
      token
    }
  }
`;

const editProfile = gql`
  mutation($username: String, $email: String, $picture: String) {
    editProfile(username: $username, email: $email, picture: $picture) {
      token
      message
    }
  }
`;

const changePassword = gql`
  mutation($oldPassword: String!, $newPassword: String!) {
    changePassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      token
      message
    }
  }
`;

const loginUser = gql`
mutation($username: String!, $password: String!) {
  loginUser(username: $username, password: $password) {
    username,
    email,
    token
  }
}
`;

const getBooks = gql`
  {
    books {
      name,
      authors,
      genre,
      year
    }
  }
`;

export {
  addUser, loginUser, getBooks, changePassword,
  verifyEmail, sendVerificationEmail,
  forgotPassword, resetPassword, editProfile
};