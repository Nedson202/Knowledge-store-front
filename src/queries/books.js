import { gql } from 'apollo-boost';

const addBook = gql`
  mutation($name: String!, $year: String!, $description: String!, $image: String, $authors: [String], $genre: [String]) {
    addBook(name: $name, description: $description, year: $year, image: $image, authors: $authors,
      genre: $genre) {
      name
    }
  }
`;

const bookFilter = gql`
  query($search: String, $from: ID, $size: ID) {
    searchBooks(searchQuery: $search, from: $from, size: $size) {
      id
      name
      genre
      authors
      averageRating
      googleAverageRating
      year
      image
      userId
      downloadable
    }
  }
`;

const fetchBook = gql`
  query($bookId: String!) {
    book(id: $bookId){
      id
      name
      genre
      authors
      averageRating
      description
      googleAverageRating
      image
      isFavorite
      downloadable
      year
      userId
      moreBooks{
        id
        name
        genre
        authors
        averageRating
        description
        googleAverageRating
        image
        downloadable
        year
        userId
      }
      reviews{
      id,
      review,
      reviewer,
      rating,
      picture
      avatarColor
      likes
      userId
      reviewsLikedBy
      bookId
      createdAt
      updatedAt
      replies {
        id
        reply
        replier
        avatarColor
        picture
        likes
        userId
        repliesLikedBy
        reviewId
        createdAt
        updatedAt
      }
      }
    }
  }
`;

const fetchUsersBooks = gql`
  {
    usersBooks{
      id
      name
      genre
      authors
      averageRating
      googleAverageRating
      year
      image
      userId
    }
  }
`;

const removeFavorites = gql`
  mutation($books: [String]){
    removeFavorites(books: $books){
      message
    }
  }
`;

const getFavorites = gql`
  {
    favoriteBooks {
      id
      name
      genre
      authors
      averageRating
      description
      googleAverageRating
      image
      downloadable
      year
      userId
    }
  }
`;

const addToFavorites = gql`
  mutation($bookId: String){
    addFavorite(bookId: $bookId){
      message
    }
  }
`;

const removeBook = gql`
  mutation($bookId: String){
    deleteBook(bookId: $bookId){
      message
    }
  }
`;

export {
  fetchBook, removeFavorites, addBook, bookFilter,
  fetchUsersBooks, removeBook, addToFavorites, getFavorites
};
