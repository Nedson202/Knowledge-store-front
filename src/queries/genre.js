import { gql } from 'apollo-boost';

const getGenres = gql`
  {
    getGenres {
      id,
      genre
    }
  }
`;

const editGenre = gql`
  mutation($reviewId: String!, $reviewEdit: String!, $rating: Float!) {
    editReview(reviewId: $reviewId, review: $reviewEdit, rating: $rating) {
      review
    }
  }
`;

const bookFilter = gql`
  query($search: String, $from: ID, $size: ID) {
    searchBooks(searchQuery: $search, from: $from, size: $size) {
      id
      name
      genre
      authors
      averageRating
      googleAverageRating
      year
      image
      userId
      downloadable
    }
  }
`;

export {
  getGenres, editGenre, bookFilter
};