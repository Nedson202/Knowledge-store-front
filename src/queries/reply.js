import { gql } from 'apollo-boost';

const addReply = gql`
  mutation($reviewId: String!, $reply: String!) {
    addReply(reviewId: $reviewId, reply: $reply) {
      reply
    }
  }
`;

const editReply = gql`
  mutation($replyId: String!, $replyEdit: String!) {
    editReply(replyId: $replyId, reply: $replyEdit) {
      reply
    }
  }
`;

const toggleLikeOnReply = gql`
  mutation($replyId: String!) {
    toggleLikeOnReply(replyId: $replyId) {
      id
    }
  }
`;

const deleteReply = gql`
  mutation($replyId: String!) {
    deleteReply(replyId: $replyId) {
      id
    }
  }
`;

export {
  addReply, editReply, deleteReply, toggleLikeOnReply
};
