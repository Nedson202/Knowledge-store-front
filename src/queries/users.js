import { gql } from 'apollo-boost';

const filterUsers = gql`
  query($type: String!) {
    fetchUsers(type: $type) {
      id
      username
      email
      role
    }
  }
`;

const toggleAdmin = gql`
  mutation($email: String!, $adminAction: String!) {
    toggleAdmin(email: $email, adminAction: $adminAction) {
      message
    }
  }
`;

export {
  toggleAdmin, filterUsers
};
