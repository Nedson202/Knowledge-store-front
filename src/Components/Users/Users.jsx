import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { compose, graphql, withApollo } from 'react-apollo';
import { Select } from 'antd';
import './_Users.scss';
import Tables from '../Table/Tables';
import { filterUsers, toggleAdmin } from '../../queries/users';
import AddAdminModal from '../AddAdminModal/AddAdminModal';
import toaster from '../../utils/toast';
import Spinner from '../Spinner/Spinner';
import errorHandler from '../../utils/errorHandler';
import modalCloser from '../../utils/modalCloser';

const { Option } = Select;

class Users extends Component {
  state = {
    filteredUsers: [],
    email: '',
  };

  handleChange = async (value) => {
    const { client } = this.props;
    client.query({
      query: filterUsers,
      variables: {
        type: value.toLowerCase()
      }
    }).then((response) => {
      const { fetchUsers } = response.data;
      if (fetchUsers.length === 0) toaster('error', 'Sorry no match was found');
      this.setState({ filteredUsers: fetchUsers || [] });
    });
  }

  handleEmailChange = (event) => {
    const { value } = event.target;
    this.setState({ email: value });
  }

  handleToggleAdmin = (adminAction, userEmail) => () => {
    const { toggleAdminQuery } = this.props;
    const { email } = this.state;
    toggleAdminQuery({
      variables: {
        email: adminAction.match('add') ? email : userEmail,
        adminAction
      },
      refetchQueries: this.refetchQuery()
    }).then((response) => {
      const { toggleAdmin: { message } } = response.data;
      modalCloser('close-user');
      this.setState({ email: '' });
      toaster('success', message);
    }).catch((error) => {
      const messages = errorHandler(error);
      messages.forEach(message => toaster('error', message));
    });
  }

  refetchQuery() {
    return [
      {
        query: filterUsers,
        variables: {
          type: 'all'
        }
      }
    ];
  }

  renderSelectField() {
    const { filteredUsers } = this.state;
    return (
      <Fragment>
        <Select defaultValue="Filter (All)" onChange={this.handleChange}>
          {filteredUsers.length && <Option value="All">All</Option>}
          <Option value="User">User</Option>
          <Option value="Admin">Admin</Option>
          <Option value="Super">Super</Option>
        </Select>
      </Fragment>
    );
  }

  renderHeader() {
    return (
      <div className="admin-panel__header">
        <h3>Users</h3>
        <div>
          {this.renderSelectField()}
          <button
            type="button"
            className="btn btn-primary btn-raised"
            data-toggle="modal"
            data-target="#AddAdminModal"
          >
            Add Admin
          </button>
        </div>
      </div>
    );
  }

  renderTable(filteredResult) {
    return (
      <Tables
        users={filteredResult}
        removeAdmin={this.handleToggleAdmin}
      />
    );
  }

  render() {
    const { fetchUsersQuery } = this.props;
    const { filteredUsers } = this.state;
    const users = fetchUsersQuery.fetchUsers !== undefined ? fetchUsersQuery.fetchUsers : [];
    const filteredResult = filteredUsers.length ? filteredUsers : users;
    return (
      <div className="admin-panel">
        {this.renderHeader()}
        {fetchUsersQuery.loading ? (
          <span className="table-spinner">
            <Spinner spinnerStyle={45} />
          </span>
        ) : this.renderTable(filteredResult)}
        <AddAdminModal
          handleInputChange={this.handleEmailChange}
          addAmin={this.handleToggleAdmin}
        />
      </div>
    );
  }
}

Users.propTypes = {
  client: PropTypes.object,
  fetchUsersQuery: PropTypes.object,
  toggleAdminQuery: PropTypes.func,
};

Users.defaultProps = {
  client: {},
  fetchUsersQuery: {},
  toggleAdminQuery: () => { },
};

export default compose(
  withApollo,
  graphql(filterUsers, {
    name: 'fetchUsersQuery',
    options: () => ({
      variables: {
        type: 'all'
      }
    })
  }),
  graphql(toggleAdmin, {
    name: 'toggleAdminQuery'
  })
)(Users);
