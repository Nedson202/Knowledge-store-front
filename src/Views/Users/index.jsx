import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { compose, graphql, withApollo } from 'react-apollo';
import { Select } from 'antd';

import Tables from '../../Components/Table';
import AddAdminModal from '../../Components/AddAdminModal';

import { filterUsers, toggleAdmin } from '../../queries/users';
import { toaster, errorHandler, modalToggler } from '../../utils';
import Spinner from '../../Components/Spinner';
import {
  SUCCESS, CLOSE_USER, ADD, TOASTR_ERROR, FETCH_USERS_QUERY,
  TOGGLE_ADMIN_QUERY,
  FILTER_USERS_QUERY
} from '../../settings';

const { Option } = Select;

class Users extends Component {
  state = {
    filteredUsers: [],
    email: '',
  };

  handleChange = async (value) => {
    const { filterUsersQuery } = this.props;

    try {
      const response = await filterUsersQuery({
        variables: {
          type: value.toLowerCase()
        }
      });

      const { fetchUsers } = response.data;

      if (fetchUsers.length === 0) {
        toaster(TOASTR_ERROR, 'Sorry no match was found');
      }

      this.setState({ filteredUsers: fetchUsers || [] });
    } catch (error) {
      console.error(error);
    }
  }

  handleEmailChange = (event) => {
    const { value } = event.target;
    this.setState({ email: value });
  }

  handleToggleAdmin = (adminAction, userEmail) => async () => {
    const { toggleAdminQuery } = this.props;
    const { email } = this.state;

    try {
      const response = await toggleAdminQuery({
        variables: {
          email: adminAction.match(ADD) ? email : userEmail,
          adminAction
        },
        refetchQueries: this.refetchQuery()
      });

      const { toggleAdmin: { message } } = response.data;
      modalToggler(CLOSE_USER);
      this.setState({ email: '' });
      toaster(SUCCESS, message);
    } catch (error) {
      const messages = errorHandler(error);
      messages.forEach(message => toaster(TOASTR_ERROR, message));
    }
  }

  refetchQuery() {
    return [
      {
        query: filterUsers,
        variables: {
          type: 'all'
        }
      }
    ];
  }

  renderSelectField() {
    const { filteredUsers } = this.state;
    return (
      <Fragment>
        <Select defaultValue="Filter (All)" onChange={this.handleChange}>
          {filteredUsers.length && <Option value="All">All</Option>}
          <Option value="User">User</Option>
          <Option value="Admin">Admin</Option>
          <Option value="Super">Super</Option>
        </Select>
      </Fragment>
    );
  }

  renderHeader() {
    return (
      <div className="admin-panel__header">
        <h3>Users</h3>
        <div>
          {this.renderSelectField()}
          <button
            type="button"
            className="btn btn-primary btn-raised"
            data-toggle="modal"
            data-target="#AddAdminModal"
          >
            Add Admin
          </button>
        </div>
      </div>
    );
  }

  renderTable(filteredResult) {
    return (
      <Tables
        users={filteredResult}
        removeAdmin={this.handleToggleAdmin}
      />
    );
  }

  render() {
    const { fetchUsersQuery } = this.props;
    const { filteredUsers } = this.state;
    const users = fetchUsersQuery.fetchUsers !== undefined
      ? fetchUsersQuery.fetchUsers : [];
    const filteredResult = filteredUsers.length ? filteredUsers : users;
    return (
      <div className="admin-panel">
        {this.renderHeader()}
        {fetchUsersQuery.loading ? (
          <span className="table-spinner">
            <Spinner spinnerStyle={45} />
          </span>
        ) : this.renderTable(filteredResult)}
        <AddAdminModal
          handleInputChange={this.handleEmailChange}
          addAmin={this.handleToggleAdmin}
        />
      </div>
    );
  }
}

Users.propTypes = {
  filterUsersQuery: PropTypes.func,
  fetchUsersQuery: PropTypes.object,
  toggleAdminQuery: PropTypes.func,
};

Users.defaultProps = {
  filterUsersQuery: () => { },
  fetchUsersQuery: {},
  toggleAdminQuery: () => { },
};

export default compose(
  withApollo,
  graphql(filterUsers, {
    name: FETCH_USERS_QUERY,
    options: () => ({
      variables: {
        type: 'all'
      }
    })
  }),
  graphql(toggleAdmin, {
    name: TOGGLE_ADMIN_QUERY
  }),
  graphql(filterUsers, {
    name: FILTER_USERS_QUERY
  }),
)(Users);
