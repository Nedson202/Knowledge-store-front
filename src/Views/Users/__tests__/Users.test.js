import React from 'react';

import {
  render, fireEvent, AllProviders, cleanup, waitForTime
} from 'test-utils';
import { filterUsers, toggleAdmin } from 'queries/users';

import Users from '..';

afterEach(cleanup);

describe('Users', () => {
  it('should render', async () => {
    const fetchUsersData = [
      {
        id: '12ERtr78h',
        username: 'Cooper AL',
        email: 'cooper.aL@al.cooper',
        role: 'user',
      }
    ];

    const mocks = [
      {
        request: {
          query: filterUsers,
          variables: {
            type: 'all'
          },
        },
        result: { data: { fetchUsers: fetchUsersData } }
      },
    ];

    const {
      container, asFragment, getByText, queryAllByText, getByTestId,
      queryByTestId,
    } = render(
      <AllProviders apolloMocks={mocks}>
        <Users />
      </AllProviders>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container).toBeVisible();

    getByText('Filter (All)');
    getByText('Users');
    queryAllByText('Add Admin');
    getByTestId('loading-indicator');

    await waitForTime(0);

    const loadingIndicator = queryByTestId('loading-indicator');
    expect(loadingIndicator).toBeNull();
  });

  it('should toggle admin', async () => {
    const toggleAdminData = {
      message: 'Admin added successfully'
    };

    const fetchUsersData = [
      {
        id: '12ERtr78h',
        username: 'Cooper AL',
        email: 'cooper.aL@al.cooper',
        role: 'admin',
      }
    ];

    const mocks = [
      {
        request: {
          query: toggleAdmin,
          variables: {
            email: 'cooper.aL@al.cooper',
            adminAction: 'add'
          },
        },
        result: { data: { toggleAdmin: toggleAdminData } }
      },
      {
        request: {
          query: filterUsers,
          variables: {
            type: 'all'
          },
        },
        result: { data: { fetchUsers: fetchUsersData } }
      },
      {
        request: {
          query: filterUsers,
          variables: {
            type: 'all'
          },
        },
        result: { data: { fetchUsers: fetchUsersData } }
      },
    ];

    const {
      getByLabelText, getByTestId, getByText
    } = render(
      <AllProviders apolloMocks={mocks}>
        <Users />
      </AllProviders>
    );

    const addAdminInput = getByLabelText('User Email');
    fireEvent.change(addAdminInput, { target: { value: 'cooper.aL@al.cooper' } });

    const addAdminButton = getByTestId('add-admin-button');
    addAdminButton.click();

    await waitForTime(0);

    getByText('Cooper AL');
    getByText('admin');
  });

  it('should trigger user filter', async () => {
    const fetchUsersData = [
      {
        id: '12ERtr78h',
        username: 'Cooper AL',
        email: 'cooper.aL@al.cooper',
        role: 'super',
      }
    ];

    const mocks = [
      {
        request: {
          query: filterUsers,
          variables: {
            type: 'super'
          },
        },
        result: { data: { fetchUsers: fetchUsersData } }
      },
    ];

    const { getByText } = render(
      <AllProviders apolloMocks={mocks}>
        <Users />
      </AllProviders>
    );

    fireEvent.click(
      getByText((_content, element) => element.getAttribute('aria-label') === 'icon: down')
    );
    const item = getByText('Super');
    fireEvent.click(item);

    await waitForTime(0);
  });

  it('should trigger user filter with no result', async () => {
    const fetchUsersData = [];

    const mocks = [
      {
        request: {
          query: filterUsers,
          variables: {
            type: 'super'
          },
        },
        result: { data: { fetchUsers: fetchUsersData } }
      },
    ];

    const { getByText } = render(
      <AllProviders apolloMocks={mocks}>
        <Users />
      </AllProviders>
    );

    fireEvent.click(
      getByText((_content, element) => element.getAttribute('aria-label') === 'icon: down')
    );
    const item = getByText('Super');
    fireEvent.click(item);

    await waitForTime(0);
  });
});
