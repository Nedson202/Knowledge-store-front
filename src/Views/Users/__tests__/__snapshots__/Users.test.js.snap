// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users should render 1`] = `
<DocumentFragment>
  <div
    class="container-content"
  >
    <div
      class="admin-panel"
    >
      <div
        class="admin-panel__header"
      >
        <h3>
          Users
        </h3>
        <div>
          <div
            class="ant-select ant-select-enabled"
          >
            <div
              aria-autocomplete="list"
              aria-controls="test-uuid"
              aria-expanded="false"
              aria-haspopup="true"
              class="ant-select-selection
            ant-select-selection--single"
              role="combobox"
              tabindex="0"
            >
              <div
                class="ant-select-selection__rendered"
              >
                <div
                  class="ant-select-selection-selected-value"
                  style="display: block; opacity: 1;"
                  title="Filter (All)"
                >
                  Filter (All)
                </div>
              </div>
              <span
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <i
                  aria-label="icon: down"
                  class="anticon anticon-down ant-select-arrow-icon"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
          <button
            class="btn btn-primary btn-raised"
            data-target="#AddAdminModal"
            data-toggle="modal"
            type="button"
          >
            Add Admin
          </button>
        </div>
      </div>
      <span
        class="table-spinner"
      >
        <div
          data-testid="loading-indicator"
        >
          <div
            class="ant-spin ant-spin-spinning"
          >
            <i
              aria-label="icon: loading"
              class="anticon anticon-loading ant-spin-dot"
              style="font-size: 45px; color: rgb(0, 92, 151);"
            >
              <svg
                aria-hidden="true"
                class="anticon-spin"
                data-icon="loading"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                />
              </svg>
            </i>
          </div>
        </div>
      </span>
      <div
        class="admin-panel"
      >
        <div
          aria-hidden="true"
          aria-labelledby="Add admin modal"
          class="modal fade"
          data-testid="add-admin-modal"
          id="AddAdminModal"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="modal-dialog modal-dialog-centered"
            role="document"
          >
            <div
              class="modal-content"
            >
              <div
                class="modal-header"
              >
                <h5
                  class="modal-title"
                >
                  Add Admin
                </h5>
                <button
                  aria-label="Close"
                  class="close"
                  data-dismiss="modal"
                  id="close-user"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    Ã—
                  </span>
                </button>
              </div>
              <div
                class="modal-body"
              >
                <form
                  class="add-admin-form"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="bmd-label-floating"
                      for="admin-email"
                    >
                      User Email
                    </label>
                    <input
                      class="form-control"
                      id="admin-email"
                      type="email"
                    />
                  </div>
                </form>
              </div>
              <div
                class="modal-footer"
              >
                <button
                  class="btn btn-default btn-raised cancel-button"
                  data-dismiss="modal"
                  type="button"
                >
                  Close
                </button>
                <button
                  class="btn btn-primary btn-raised text-case add-admin"
                  data-testid="add-admin-button"
                  type="button"
                >
                  Add Admin
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
